File: lib/models/wifi_network.dart
class WifiNetwork {
  final String ssid;
  final int rssi;

  WifiNetwork({required this.ssid, required this.rssi});

  factory WifiNetwork.fromMap(Map<dynamic, dynamic> map) {
    return WifiNetwork(
      ssid: map['ssid'] as String,
      rssi: map['rssi'] as int,
    );
  }
}

File: android/src/main/kotlin/how/virc/flutter_esp_ble_prov/WifiNetwork.kt
// src/main/kotlin/how/virc/flutter_esp_ble_prov/WifiNetwork.kt

package how.virc.flutter_esp_ble_prov

import java.io.Serializable

/**
 * Data class representing a Wi-Fi network with SSID and RSSI.
 */
data class WifiNetwork(
    val ssid: String,
    val rssi: Int
) : Serializable
